src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/ExceptionRecordCreatorTest.java

```diff
@@ -0,0 +1,127 @@
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.getRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.givenThat;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static org.awaitility.Awaitility.await;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.fileContentAsString;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.config.Environment.CASE_REF;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.config.Environment.CASE_TYPE_BULK_SCAN;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.config.Environment.CASE_TYPE_EXCEPTION_RECORD;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.config.Environment.GET_CASE_URL;
+
+@ExtendWith(SpringExtension.class)
+@IntegrationTest
+class ExceptionRecordCreatorTest {
+
+    private static final String CASE_EVENT_TRIGGER_START_URL = Environment.CASE_EVENT_TRIGGER_START_URL
+        .replace(CASE_TYPE_BULK_SCAN, CASE_TYPE_EXCEPTION_RECORD);
+    private static final String CASE_SUBMIT_URL = Environment.CASE_SUBMIT_URL
+        .replace(CASE_TYPE_BULK_SCAN, CASE_TYPE_EXCEPTION_RECORD);
+
+    @Autowired
+    @Lazy
+    private WireMockServer server;
+
+    @Autowired
+    @Lazy
+    private MessageSender messageSender;
+
+    @BeforeEach
+    void before() {
+        WireMock.configureFor(server.port());
+
+        givenThat(get(GET_CASE_URL).willReturn(aResponse().withStatus(HttpStatus.NOT_FOUND.value())));
+        givenThat(get(CASE_EVENT_TRIGGER_START_URL).willReturn(aResponse().withBody(
+            "{\"case_details\":null,\"event_id\":\"eid\",\"token\":\"etoken\"}"
+        )));
+    }
+
+    @DisplayName("Should create exception record for supplementary evidence when case record is not found")
+    @Test
+    void should_create_exception_record_for_supplementary_evidence_when_case_record_is_not_found() {
+        messageSender.send(messageFromFile("supplementary-evidence-example.json"));
+
+        await()
+            .atMost(30, TimeUnit.SECONDS)
+            .ignoreExceptions()
+            .until(() -> {
+                server.verify(getRequestedFor(urlPathEqualTo(GET_CASE_URL)));
+                server.verify(postRequestedFor(urlPathEqualTo(CASE_SUBMIT_URL)));
+                return true;
+            });
+    }
+
+    @DisplayName("Should create exception record for supplementary evidence when case ref is not provided")
+    @Test
+    void should_create_exception_record_for_supplementary_evidence_when_case_ref_is_not_provided() {
+        Message incompleteSupplementaryMessage = new Message(fileContentAsString(
+            "servicebus/message/supplementary-evidence-example.json"
+        ).replace(CASE_REF, ""));
```

Comment:
can be null instead

Reply:
in theory it can. other `message` samples in there have `null` - so it is covering edge case. bot null and empty string are handled same way